kernel/src/hil/qdec.rs

```diff
@@ -0,0 +1,33 @@
+//! Interface for a Qdec compatible chip
+//!
+//! This trait provides a stanfard interface for chips with a
+//! quadrature decoder. Note this interface is experimental and
+//! may need further updates once implemented on additional chips
+
+use crate::returncode::ReturnCode;
+
+pub trait QdecDriver {
+    /// Sets the client which will receive interrupts
+    fn set_client(&self, client: &'static dyn QdecClient);
+
+    /// Enables the SAMPLERDY interrupt
```

Comment:
I think this is too chip-specific for a HIL. What does this interrupt do? When does it fire? Can multiple chips reasonably support this?

Reply:
Thanks for comment! I will update the comments/function names to prevent the HIL from being too closely tied with the nRF. I am currently looking into whether the current QDEC setup is compatible with other chips.